context/PoolContext/index.tsx

```diff
@@ -250,6 +250,7 @@ export const PoolStore: React.FC<Children> = ({ children }: Children) => {
                 PoolCommitter__factory.abi,
                 subscriptionProvider,
             ) as PoolCommitter;
+            provider.removeAllListeners();
```

Comment:
would this line here also remove the listener on the keeperInstance? if so we might not need the other logic

Reply:
actually we might not want this? would this stop listening to events on subsequent calls to this function? ie if we called this once for pool A and then once for pool B, when called for pool B it would unsub from the events that were set up when it was called for pool A?

Reply:
this is a mistake